swift/codegen/schema.yml

```diff
@@ -28,9 +28,12 @@ Location:
   end_line: int
   end_column: int
 
-Type:
-  diagnostics_name: string
-  canonical_type: Type
+Type: {}
+#  diagnostics_name: string
```

Comment:
I think this might be related to the `static_assert` in `SwiftDispatcher`?

Reply:
TL;DR is that we only need to comment out class properties of final classes, but it was more systematic (and easier with regexp edits) to comment out all properties, and bring them back in the upcoming PRs.

The more detailed explanation follows.

In order to emit a correctly typed `@` trap ids (say for example `@concrete_var_decl`), we need to emit the corresponding "binding" trap entry (`concrete_var_decls`). Now if this had a `@` ref column, we would need to fill that with a valid label. So for example if we have
```
my_entries(unique int id: @my_entry, int prop: @other ref);
```
which generates trap entry
```cpp
strut MyEntriesTrap { TrapLabel<MyEntryTag> id{}; TrapLabel<OtherTag> prop{}; }
```
and we emit `MyEntriesTrap{label}`, this will result in a trap emission like
```
my_entries(#1a, #ffffffff)
```
and the trap importer will silently ignore all entries using undefined labels (took me a while to diagnose). It would be ok to have fields in non-binding traps (for example those related to abstract classes) because we just do not emit those for unknown entities, and it would also be ok for non-class properties to be present (because default initialization would not pose problems for the trap importer). But I just took an easier restriction to check, and removed all properties from schema as it was easier.

Reply:
By the way, in an upcoming PR I added an `assert` checking that we never emit an undefined label, which will catch those unfortunate default initialization of labels. We can rework the `assert` later with a proper diagnostics module. Another mitigation against the risk of using default-initialized labels could be to delete the default constructor of trap entries and only have the constructors with the full number of fields initialized.